relay-server/src/actors/processor.rs

```diff
@@ -2358,14 +2364,68 @@ impl EnvelopeProcessorService {
     }
 
     /// Computes the sampling decision on the incoming transaction.
-    fn compute_sampling_decision(&self, state: &mut ProcessEnvelopeState) {
-        state.sampling_result = utils::get_sampling_result(
-            self.inner.config.processing_enabled(),
-            Some(&state.project_state),
-            state.sampling_project_state.as_deref(),
-            state.envelope().dsc(),
-            state.event.value(),
-        );
+    fn compute_sampling_decision(
```

Comment:
removed the `&self` because it was only used to check if processing is enabled, which also made it way easier to unit test. 

Not sure if it should still be under `EnvelopeProcessorService` though, how about we move it to `utils` like the `utils::get_sampling_result`?

Reply:
I'd suggest to revert this change for three reasons:
 1. The parameter list of this function is long. That was also the reason why `ProcessEnvelopeState` was introduced in the first place, to pass processing information together. 
 2. We will soon need more fields from the state, such as access to Redis as well as the `received` timestamp (it is currently a bug that this function uses `Utc::now()`).
 3. Associated functions like this one that neither work with the type that they are declared on, nor return it, are unidiomatic.

Reply:
alright, i'll revert it, although personally I prefer it like this. My thinking was that we would clearly delineate the `process_state` function which runs entirely on side effects, and it would in turn call pure functions, meaning we would avoid passing in `state` any further from `processing_state`. 

I'd say the long parameter list might tell us something about this function that simply passing in state would mask. Passing in `state` here feels a lot like using global variables. 

in the function (as expressed by the first early return) you can see how 'event' and 'sampling_config' need each other, same with dsc and root_sampling_config. Perhaps we could explore a newtype that encapsulates the "Getter" and a sampling config in a later PR? 

3:  I agree its not idiomatic, so i think it would be good in `utils` of the same crate

Reply:
ok so, reverting it leads to complicating many tests I wrote that relied on its functional approach, where you have to start the service, create a mock `processenvelopestate` and `projectstate` to check if a field on processenvelopestate has changed. 
 
could you verify again if you think I should revert this? my suggestion would be to keep it functional, move it to utils, and let the long param list stand as a sign for us to think about refactoring in the future (rather than hiding the params in state).

Reply:
Makes sense. Given this is an internal function of the processor, we can easily revisit this at a later point. There's limited exposure of this API. Let's move forward.